iOS build error (Undefined symbols for architecture arm64)

**Godot Version**
3.0.2

**OS Version**
macOS High Sierra 10.13.4

**Issue**
produces this error when tries to build on phone
```
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_SKProduct", referenced from:
      l_OBJC_$_CATEGORY_SKProduct_$_LocalizedPrice in ios.a(in_app_store.iphone.opt.debug.arm64.o)
  "_OBJC_CLASS_$_SKPayment", referenced from:
      objc-class-ref in ios.a(in_app_store.iphone.opt.debug.arm64.o)
  "_OBJC_CLASS_$_SKPaymentQueue", referenced from:
      objc-class-ref in ios.a(in_app_store.iphone.opt.debug.arm64.o)
  "_OBJC_CLASS_$_SKProductsRequest", referenced from:
      objc-class-ref in ios.a(in_app_store.iphone.opt.debug.arm64.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
````

**How to reproduce**
the build works fine in a blank projects with no scenes, produces error when added a scene

sorry i'm not familiar with iOS, xcode and can't get help from Q&A, discord and forums

--- a/modules/arkit/SCsub
+++ b/modules/arkit/SCsub
@@ -5,6 +5,9 @@ Import("env_modules")
 
 env_arkit = env_modules.Clone()
 
+# (iOS) Enable module support
+env_arkit.Append(CCFLAGS=["-fmodules", "-fcxx-modules"])
+
 # (iOS) Build as separate static library
 modules_sources = []
 env_arkit.add_source_files(modules_sources, "*.cpp")

--- a/modules/camera/SCsub
+++ b/modules/camera/SCsub
@@ -6,6 +6,9 @@ Import("env_modules")
 env_camera = env_modules.Clone()
 
 if env["platform"] == "iphone":
+    # (iOS) Enable module support
+    env_camera.Append(CCFLAGS=["-fmodules", "-fcxx-modules"])
+
     # (iOS) Build as separate static library
     modules_sources = []
     env_camera.add_source_files(modules_sources, "register_types.cpp")

--- a/platform/iphone/SCsub
+++ b/platform/iphone/SCsub
@@ -19,6 +19,9 @@ iphone_lib = [
 env_ios = env.Clone()
 ios_lib = env_ios.add_library("iphone", iphone_lib)
 
+# (iOS) Enable module support
+env_ios.Append(CCFLAGS=["-fmodules", "-fcxx-modules"])
+
 
 def combine_libs(target=None, source=None, env=None):
     lib_path = target[0].srcnode().abspath

--- a/platform/iphone/export/export.cpp
+++ b/platform/iphone/export/export.cpp
@@ -343,6 +343,9 @@ void EditorExportPlatformIOS::_fix_config_file(const Ref<EditorExportPreset> &p_
 		} else if (lines[i].find("$push_notifications") != -1) {
 			bool is_on = p_preset->get("capabilities/push_notifications");
 			strnew += lines[i].replace("$push_notifications", is_on ? "1" : "0") + "\n";
+		} else if (lines[i].find("$entitlements_push_notifications") != -1) {
+			bool is_on = p_preset->get("capabilities/push_notifications");
+			strnew += lines[i].replace("$entitlements_push_notifications", is_on ? "<key>aps-environment</key><string>development</string>" : "") + "\n";
 		} else if (lines[i].find("$required_device_capabilities") != -1) {
 			String capabilities;
 
@@ -932,6 +935,7 @@ Error EditorExportPlatformIOS::export_project(const Ref<EditorExportPreset> &p_p
 	files_to_parse.insert("godot_ios/dummy.cpp");
 	files_to_parse.insert("godot_ios.xcodeproj/project.xcworkspace/contents.xcworkspacedata");
 	files_to_parse.insert("godot_ios.xcodeproj/xcshareddata/xcschemes/godot_ios.xcscheme");
+	files_to_parse.insert("godot_ios/godot_ios.entitlements");
 
 	IOSConfigData config_data = {
 		pkg_name,

